Apps/Database/src/Delegates/DbApplicationSettingsDelegate.cs

```diff
@@ -69,33 +62,11 @@ public void DeleteApplicationSetting(ApplicationSetting appSetting)
         }
 
         /// <inheritdoc/>
-        public IList<ApplicationSetting> GetApplicationSettings(string application, string component)
-        {
-            this.logger.LogTrace("Getting application setting for {Application}/{Component} from DB...", application, component);
-            IList<ApplicationSetting> retVal = this.dbContext.ApplicationSetting
-                .Where(
-                    p => p.Application == application &&
-                         p.Component == component)
-                .ToList();
-            this.logger.LogDebug("Finished getting application setting for {Application}/{Component} from DB...", application, component);
-
-            return retVal;
-        }
-
-        /// <inheritdoc/>
-        public async Task AddApplicationSettingAsync(ApplicationSetting appSetting, CancellationToken ct = default)
+        public void AddApplicationSetting(ApplicationSetting appSetting)
```

Comment:
Is async unnecessary for Add?

Reply:
Yes, because `dbContext.Add()` (along with `dbContext.Update()` and `dbContext.Remove()`) do not commit changes to the database.

The INSERT/UPDATE/DELETE operations are performed by calling `dbContext.SaveChanges()` after.

There does exist a `dbContext.AddAsync()` method, but its documentation is clear that it shouldn't be used as a general replacement:

> This method is async only to allow special value generators, such as the one used by 'Microsoft.EntityFrameworkCore.Metadata.SqlServerValueGenerationStrategy.SequenceHiLo', to access the database asynchronously. For all other cases the non async method should be used.